core/dnsserver/register.go

```diff
@@ -124,6 +124,49 @@ func (c *Config) AddMiddleware(m middleware.Middleware) {
 	c.Middleware = append(c.Middleware, m)
 }
 
+// RegisterHandler adds a handler to a site's handler registration. Handlers
+// should use this if the want to announce that they exist to other middleware.
+func (c *Config) RegisterHandler(h middleware.Handler) {
+	if c.Registry == nil {
+		c.Registry = make(map[string]middleware.Handler)
+	}
+
+	if _, ok := c.Registry[h.Name()]; ok {
+		panic("foot!")
```

Comment:
We can have multiple instances of the same mw in a given chain (e.g., rewrite). Not an issue now but maybe that's something we need to deal with here.